test/blocks/global-footer/global-footer.test.js

```diff
@@ -0,0 +1,247 @@
+import { expect } from '@esm-bundle/chai';
+import sinon, { stub } from 'sinon';
+import {
+  allElementsVisible,
+  visibleSelectorsDesktop,
+  visibleSelectorsMobile,
+  containerSelector,
+  createFullGlobalFooter,
+  insertDummyElementOnTop,
+  isElementVisible,
+  waitForFooterToDecorate,
+  allSelectors,
+} from './test-utilities.js';
+import baseFooter from './mocks/base-footer.js';
+import fetchedFooter from './mocks/fetched-footer.js';
+import icons from './mocks/icons.js';
+import { mockRes } from '../global-navigation/test-utilities.js';
+import placeholders from './mocks/placeholders.js';
+
+const originalFetch = window.fetch;
+
+describe('global footer', () => {
+  let clock = null;
+  beforeEach(async () => {
+    document.body.innerHTML = baseFooter;
+    clock = sinon.useFakeTimers({
+      toFake: ['setTimeout'],
+      shouldAdvanceTime: true,
+    });
+
+    window.fetch = stub().callsFake((url) => {
+      if (url.includes('/footer')) {
+        return mockRes({
+          payload: fetchedFooter(
+            { regionPickerHash: '/fragments/regions#langnav' },
+          ),
+        });
+      }
+
+      if (url.includes('/placeholders')) return mockRes({ payload: placeholders });
+
+      if (url.includes('icons.svg')) return mockRes({ payload: icons });
+
+      return null;
+    });
+  });
+
+  afterEach(() => {
+    clock.restore();
```

Comment:
Should we also clear the body content here to ensure we're not polluting any other tests after the footer suite is completed?

Reply:
Yes, makes sense